### PR TITLE
Mentionrole update to allow posting in specified channel and better role searching.

### PR DESCRIPTION
**Pull request by Satty9361**
Added a new flag to allow posting message in a different channel. 

Modified role searching to try to search by role ID if match using name is not found.

### CODE REVIEW

File: stdcommands/mentionrole/mentionrole.go [hunk 1]
```diff
@@ -74,27 +76,52 @@ func cmdFuncMentionRole(data *dcmd.Data) (interface{}, error) {
...
-		return "No role with the name `" + data.Args[0].Str() + "` found", nil
+		return "No role with the name or ID`" + roleS + "` found", nil
+	}
+	
+	cID := data.CS.ID
+	c := data.Switch("channel")
+	if c.Value != nil {
+		cID = c.Value.(*dstate.ChannelState).ID
+		
+		perms, err := data.GS.MemberPermissions(true, cID, data.Msg.Author.ID)
```
Comment (jonas747) [inline stdcommands/mentionrole/mentionrole.go]: set the first argument to false since were holding a read lock on the state already

Comment (Satty9361) [reply stdcommands/mentionrole/mentionrole.go]: I see. Changed lock to false .

**Comment by jonas747** [comment]
I don't see any changes made to the command struct, like the channel switch, did you forget it?

**Comment by Satty9361** [comment]
Yeah Yikes! I did forget adding the switch when generating the PR. Made the changes.

**Comment by jonas747** [comment]
noice

**Pull request merged**